pkg/common/dtypeutils/dtypeutils.go

```diff
@@ -657,3 +657,42 @@ type LogFileData struct {
 	StatusCode  int
 	Duration    int64
 }
+
+func CompareValues(leftVal interface{}, rightVal interface{}, Op string) bool {
+	switch Op {
+	case "=":
+		return leftVal == rightVal
+	case "!=":
+		return leftVal != rightVal
+	}
+
+	leftFloatVal, errLeft := ConvertToFloat(leftVal, 64)
+	rightFloatVal, errRight := ConvertToFloat(rightVal, 64)
+
+	if errLeft == nil || errRight == nil {
+		switch Op {
+		case "<":
+			return leftFloatVal < rightFloatVal
```

Comment:
Should `errLeft == nil || errRight == nil` be `errLeft == nil && errRight == nil`? Or if we keep `||` and e.g., `errLeft != nil`, should we set `leftFloatVal = 0`?

Reply:
ohh, good catch, it is `&&` not `||`